bin/cleanup_old_instances.rb

```diff
@@ -1,12 +1,15 @@
 target_date = 3.months.ago
-old_instances = Kuroko2::JobInstance
+count = 0
+
+Kuroko2::JobInstance
   .where('finished_at < ?', target_date)
   .or(Kuroko2::JobInstance.where('canceled_at < ?', target_date))
-
-count = old_instances.count
-
-Kuroko2::JobInstance.transaction do
-  old_instances.destroy_all
+  .order(id: :asc)
+  .in_batches do |old_instances|
+  count += old_instances.size
+  Kuroko2::JobInstance.transaction do
+    old_instances.destroy_all
+  end
```

Comment:
Is not `sleep` necessary?

Reply:
first, destroy_all is just call destroy to each instance.
https://github.com/rails/rails/blob/008538081b87940ad1a4fd29df730ec5d0421ee5/activerecord/lib/active_record/relation.rb#L379-L381
second, destroy seems to use the transaction which includes only itself.
https://github.com/rails/rails/blob/6aa5cf03ea8232180ffbbae4c130b051f813c670/activerecord/lib/active_record/persistence.rb#L336-L347

Based on these, we might have 2 options:

- Just unwrap `Kuroko2::JobInstance.transaction`
- `old_instances.each { |i| Kuroko2::JobInstance.transaction { i.destroy } }`
  - I'm doubt to do it worth 🤔 

how do you think? @eisuke

Reply:
I think that "Just unwrap Kuroko2::JobInstance.transaction" is good.

Reply:
https://github.com/cookpad/kuroko2/pull/109 Ready for review XD